lib/SIL/Parser/ParseSIL.cpp

```diff
@@ -46,6 +48,79 @@ using namespace swift::syntax;
 //===----------------------------------------------------------------------===//
 
 namespace swift {
+
+template <class... DiagArgs, class... Args>
+InFlightDiagnostic SILParsedLocalValueMap::diagnose(SourceLoc loc,
+                                                    Diag<DiagArgs...> diagID,
+                                                    Args &&... args) {
+  return module.getASTContext().Diags.diagnose(
+      loc, Diagnostic(diagID, std::forward<Args>(args)...));
+}
+
+SILValue SILParsedLocalValueMap::getLocalValue(UnresolvedValueName name,
+                                               SILType type, SILLocation loc,
+                                               SILBuilder &builder) {
+  if (name.isUndef())
+    return SILUndef::get(type, builder.getFunction());
```

Comment:
Why not just pass in the function instead of passing in the SILBuilder. In general passing around SILBuilder is an old style that we do not want to propagate unless we really need to.

Reply:
Good to know. Done.

Reply:
Actually, I remember now, when I originally wrote this patch I did try to make this a function. Unfortunately, the builder doesn't always have a function associated with it. We just hope that if we ever run into an `undef` or forward ref, there is a function (which is an error waiting to happen). I will fix this in another PR (because it requires a larger fix).